vizro-core/src/vizro/models/_components/graph.py

```diff
@@ -71,6 +75,39 @@ def __getitem__(self, arg_name: str):
             return self.type
         return self.figure[arg_name]
 
+    # Interaction methods
+    def _get_figure_interaction_input(self) -> Dict[str, State]:
+        """Required properties when using pre-defined `filter_interaction`."""
+        return {
+            "clickData": State(component_id=self.id, component_property="clickData"),
+            "modelID": State(component_id=self.id, component_property="id"),  # required, to determine triggered model
+        }
+
+    def _filter_interaction(
+        self, data_frame: pd.DataFrame, target: str, ctd_filter_interaction: Dict[str, CallbackTriggerDict]
+    ) -> pd.DataFrame:
+        """Function to be carried out for pre-defined `filter_interaction`."""
+        ctd_click_data = ctd_filter_interaction["clickData"]
+        if not ctd_click_data["value"]:
+            return data_frame
+
+        source_graph_id: ModelID = ctd_click_data["id"]
+        source_graph_actions = _get_component_actions(model_manager[source_graph_id])
+        try:
+            custom_data_columns = model_manager[source_graph_id]["custom_data"]
+        except KeyError as exc:
+            raise KeyError(f"No `custom_data` argument found for source graph with id {source_graph_id}.") from exc
+
+        customdata = ctd_click_data["value"]["points"][0]["customdata"]
+
+        for action in source_graph_actions:
+            if action.function._function.__name__ != "filter_interaction" or target not in action.function["targets"]:
+                continue
+            for custom_data_idx, column in enumerate(custom_data_columns):
+                data_frame = data_frame[data_frame[column].isin([customdata[custom_data_idx]])]
+
+        return data_frame
```

Comment:
It looks like `_filter_interaction` method is pretty similar in Graph/Table/Grid model. What do you think about the idea of implementing `_extract_filtering_info` method per model instead (e.g. it can just return real selected `value` and maybe the `column` that has to be filtered), and conduct filtering inside the `_actions_utils._apply_filter_interaction`?

Reply:
That is an interesting idea! I think we should definitely discuss it. I guess the pros are:
- simplified and shortened code in the respective models
- avoiding repretition

The cons I can think of are:
- potentially wrong abstraction level given that e.g. the graph currently filters multiple columns, which the table/grid don't
- like this we are giving the user less control over the filtering

Open to both really :)

Reply:
I agree that it is not easy to make this kind of abstraction in general, hence I call it - "actions_info" - the biggest open question we currently have.

If we make this kind of abstraction properly, we could say that the "filter_interaction" is just the same as the "_filter" action, which sounds perfectly correct to me. 

From top of my head: All "actions_info" we need to set for every Vizro model (e.g. Button, Card, Dropdown, Graph, Grid...) is:
- `_get_input_components` - e.g. {model_id: State(...), derived_viewport_data: State(...), active_cell: State(...)} OR {model_id: State(...), value: State(...)};
- `_extract_value_from_input` - some `function`,
- `_get_output_components` - e.g. {model_id: Output(...)}  

It means that every Vizro model will have implemented these methods in a form of (`class methods`, or some `actions_info: Dict[str, any]` argument). Also, these methods/arguments could be overwritten in case of custom component creation.

Reply:
I would suggest we leave it as is for now and come back to it later. This is still a big open question indeed, and I think "Duplication is cheaper than the wrong abstraction" applies in this case.

Generally I think I agree with @petar-qb's idea, but there's still a lot of moving pieces here so hard to tell right now what the right solution is. So for now I think just focus on:
* getting these models working correctly
* initial actions v2 work to make those action `CapturedCallable` classes

And after that we can figure out what to do with all this actions info stuff.

Let's unresolve this conversation after merging though so we can easily refer back to it.